src/Stratis.Bitcoin.Features.Wallet/Controllers/WalletController.cs

```diff
@@ -253,6 +254,76 @@ public IActionResult Recover([FromBody]WalletRecoveryRequest request)
             }
         }
 
+        /// <summary>
+        /// Recovers a wallet using only the extended public key.
+        /// </summary>
+        /// <param name="request">The object containing the parameters used to recover a wallet.</param>
+        /// <returns></returns>
+        [Route("recover-via-extpubkey")]
+        [HttpPost]
+        public IActionResult RecoverViaExtPubKey([FromBody]WalletExtPubRecoveryRequest request)
+        {
+            Guard.NotNull(request, nameof(request));
+
+            if (!this.ModelState.IsValid)
+            {
+                return BuildErrorResponse(this.ModelState);
+            }
+
+            try
+            {
+                string accountExtPubKey = this.SanitiseExtPubKey(request.ExtPubKey);
+
+                Wallet wallet = this.walletManager.RecoverWallet(request.Name, ExtPubKey.Parse(accountExtPubKey), request.AccountIndex, request.CreationDate);
```

Comment:
`Wallet wallet =` is redundant

Reply:
cheers

Reply:
some other Post methods have it too, but I removed.

Reply:
Nice 🙂